### DISCUSSION TITLE
Implementing  Modal

### DISCUSSION THREAD

**Discussion by nlmf123** [django-commons/django-unicorn#444] [Q&A] [answered]
Hi, good day/. I am new here @ django-unicorn. i just wanna ask how you guys implement a simple modal using django-unicorn?

**Comment by MJPawluk** [comment, marked as answer]
Hi [nlmf123](https://github.com/nlmf123)!

I used [bulma.io](https://bulma.io) which has a [modal](https://bulma.io/documentation/components/modal/) component. To activate the component you have to add `is-active` to the `class` attribute of the modal.


For the .py file of the component I did something like this:

    # modal.py
    from django_unicorn.components import UnicornView
    
    
    class ModalView(UnicornView):
        is_active = ""
    
        def toggle_modal(self):
            if self.is_active == "":
                self.is_active = "is-active"
            else:
                self.is_active = ""

And in the .html file for the component:

    
    <div>
      <button unicorn:click="toggle_modal()"></button>
        
      <div class="modal {{ is_active }}">
              
      </div>
    </div>
